Swift push docs (#415)
WalletConnect/walletconnect-docs: docs/swift/push/dapp-usage.md

# Dapp Usage

> ### Initial configurations

Reviewer: ```suggestion
### Initial Configurations
```

Make sure what you properly configure Networking and Pair Clients first
- [Networking](../core/networking-configuration.md)
- [Pairing](../core/pairing-usage.md)

### Subscribe for Push publishers
When your `Push` dapp instance receives push request or push message from a peer client, it will publish a related event. Subscribe to publishers to reveive the requests.

```swift
Push.wallet.responsePublisher
    .receive(on: DispatchQueue.main)
    .sink { [unowned self] id, result in
        //handle event
    }.store(in: &publishers)
```
Following publishers are available to subscribe:

```swift
public var responsePublisher: AnyPublisher<(id: RPCID, result: Result<PushSubscription, PushError>), Never> 
public var deleteSubscriptionPublisher: AnyPublisher<String, Never> 
```

### Send Push Subscription:

Once you have an active pairing with a wallet, you are able to send a push subscription requests on a pairing topic.

```swift
try await Push.dapp.request(account: Account, topic: String)
```

> ### Send a push message to a wallet

Reviewer: ```suggestion
### Send a Push Message to a Wallet
```

After push subscription is established, you are able to send a PN to a wallet on a push subscription topic.

```swift
try await Push.dapp.notify(topic: String, message: PushMessage)
```

### Get active subscriptions

```swift 
Push.wallet.getActiveSubscriptions()
```

> ### Where to go from here

Reviewer: ```suggestion
### Where to Go from Here
```

> - Try our Wallet app that is part of WalletConnectSwiftV2 repository.

Reviewer: ```suggestion
- Try our [Wallet App](https://github.com/WalletConnect/WalletConnectSwiftV2/tree/main/Example/WalletApp) and to test notifications.
```
